### DISCUSSION TITLE
Blocking all merges to a branch

### DISCUSSION THREAD

**Discussion by kylekthompson** [community/community#22107] [New to GitHub]
<p>Hi,</p>
<p>I was looking through the branch restrictions for repositories in an organization to see whether it was possible to block merges to a certain branch.</p>
<p>Right now, it seems like it’s possible to do one of the following:</p>
<ol>
<li>Block <em>pushes</em> to a branch (this only affects pushing from a local copy of the repository to the branch on GitHub)</li>
<li>Block merges to a branch <em>if there are not enough reviews or if a status check has not passed</em>, but at least one of these qualifiers is required</li>
</ol>
<p>For some context, we’re building a small merge queue to manage our growing team and want to ensure all merges go through that (essentially, block pushes <em>and merges</em> from everyone except a GitHub App). Am I missing an option to entirely protect a branch from any changes, or is this not something that’s possible today?</p>
<p>Thanks!</p>

**Comment by ghost** [comment, marked as answer]
<p>Hi <a class="mention" href="/kylekthompson">@kylekthompson</a> - welcome to the GitHub Community.</p>
<p>You are correct in that it’s not possible to entirely prevent all merges on a specified branch.</p>
<p>It’s only possible to restrict the conditions in which a merge can take place using the existing branch protection settings within GitHub.</p>
<p>Sorry this isn’t the answer you are looking for!</p>

**Comment by kylekthompson** [comment]
<p>No worries, thanks for taking the time to respond!</p>

**Comment by DanielTomlinsonSage** [comment]
This needs to be a feature, somehow. You need to have certain branches that can not be overwritten, except via admin.

**Comment by lackovic** [comment]
This would come useful when the deployment pipeline is broken and we want to prevent further merges to `main` until the issues are resolved.
